src/adapter/sources.ts

```diff
@@ -355,29 +369,29 @@ export class SourceContainer {
     if (!map)
       return [];
     const sourceMapUiLocation = this._sourceMappedUiLocation(uiLocation, map);
-    if (!sourceMapUiLocation)
+    if (!isUiLocation(sourceMapUiLocation))
       return [];
 
     const r = this.getSourceMapUiLocations(sourceMapUiLocation);
     r.push(sourceMapUiLocation);
     return r;
   }
 
-  _sourceMappedUiLocation(uiLocation: UiLocation, map: SourceMap): UiLocation | undefined {
+  _sourceMappedUiLocation(uiLocation: UiLocation, map: SourceMap): UiLocation | UnmappedReason {
```

Comment:
This is awkward, let me know if you have a better idea

Reply:
I think this is fine. Throwing some typed error would be traditional OO thing to do but I don't mind this; it forces consumers to handle failures explicitly.

Reply:
Yeah and since I need to keep the failure type around later on PreferredUiLocation, I wanted to just introduce it here.